enterprise/signals.py

```diff
@@ -148,13 +156,17 @@ def assign_or_delete_enterprise_admin_role(sender, instance, **kwargs):     # py
             pending_enterprise_admin_user = None
 
         if kwargs['created'] and pending_enterprise_admin_user:
+            # Create the user in third party analytics application
+            create_tableau_user(instance.user.username, instance)
             # EnterpriseCustomerUser record was created and a pending admin user
             # exists, so assign the enterprise_admin role.
             pending_enterprise_admin_user.activate_admin_permissions(
                 user=instance.user,
                 enterprise_customer=instance.enterprise_customer,
             )
         elif not kwargs['created'] and not instance.linked:
+            # Delete the user in third party analytics application
+            delete_tableau_user(instance)
```

Comment:
Please add details regarding 
1. why are deleting tableau here, and
2. why exactly we need to delete the user?

Reply:
Done